R/IOHproblem.R

```diff
@@ -0,0 +1,79 @@
+#' S3 print function for IOHproblem
+#'
+#' @param x The IOHproblem to print
+#' @param ... Arguments for underlying function
+#'
+#' @export
+print.IOHproblem <- function(x, ...) {
+  cat(as.character.IOHproblem(x, ...))
+}
+
+#' S3 generic cat function for IOHproblem
+#'
+#' @param x The IOHproblem to print
+#'
+#' @export
+cat.IOHproblem <- function(x) cat(as.character(x))
+
+#' S3 generic as.character function for IOHproblem
+#'
+#' @param x The IOHproblem to print
+#' @param ... Arguments for underlying function
+#'
+#' @export
+as.character.IOHproblem <- function(x, ...) {
+  sprintf('IOHproblem (Instance %d of function %d %dD)', x$instance, x$function_id, x$dimension)
+}
+
+#' Get the next function of the currently initialized suite
+#'
+#' @param experimenter The IOHexperimenter object
+#'
+#' @return An IOHproblem object
+#' @export
+next_problem <- function(experimenter) {
+  ans <- c_get_next_problem()
+  if (is.null(ans) || is.null(ans$problem)) return(NULL)
+
+  return(structure(
+    list(
+      dimension = ans$dimension,
+      function_id = ans$problem,
+      instance = ans$instance,
+      fopt = c_get_fopt(),
+      xopt = c_get_xopt(),
+      params.track = experimenter$param.track,
+      obj_func = function(x) {
+        stopifnot( all(x %in% c(0,1) ))
+        if (is.null(dim(x))) x <- t(x)
+        if (ncol(x) != ans$dimension) x <- t(x)
+
+        stopifnot(ncol(x) == ans$dimension)
+        apply(x, 1, c_eval)
+      },
+      target_hit = function() {
+        c_is_target_hit()
+      },
+      set_parameters = function(param_vals){
```

Comment:
This function sets the parameter we want to track, right? Do we also set the parameter to track by default for all problems, given a parameter list is provided when creating the IOHexperimenter class?

Reply:
The way it is organized right now is as follows: when creating an IOHexperimenter class, you can specify which parameters to track. 
These can then be set in the IOHproblem classes derived from this. 
The default for this is no parameter tracking, and even if we were to set a default, the user would still need to manually call the set_parameter function in their code.